include/envoy/http/access_log.h

```diff
@@ -114,6 +114,16 @@ class RequestInfo {
    * Set whether the request is a health check request or not.
    */
   virtual void healthCheck(bool is_hc) PURE;
+
+  /**
+   * Set the downstream address
+   */
+  virtual void setDownstreamAddress(const std::string& address) PURE;
```

Comment:
@RomanDzhabarov i think @mattklein123  is suggesting to move it out of connection_manager_impl itself (it was already present) and just keep it in the request_info because the connection_manager_impl has a reference to request_info as well.